react-native/react/menubar/index.render.desktop.js

```diff
@@ -79,34 +94,159 @@ class OpeningMessage extends Component {
   }
 }
 
+class FolderRow extends Component {
+  props: {
+    username: string,
+    folder: FolderInfo
+  };
+
+  renderFolderText (text, color, key = null) {
+    return <div key={key || text} style={{...personalTLDStyle, color}}>{text}</div>
+  }
+
+  // Folder name is rendered a bit weird. If we only have our name it's the normal color
+  // If we have our name and someone else, our name is toned down
+  renderFolderName () {
+    const {username, folder: {folderName, openFolder}} = this.props
+    let folderText
+
+    if (username === folderName) {
+      folderText = [this.renderFolderText(username, colors.lightBlue)]
+    } else {
+      let users = canonicalizeUsernames(username, parseFolderNameToUsers(folderName))
+
+      folderText = users.map(u => {
+        if (u === username) {
+          return this.renderFolderText(u, colors.lightTeal)
+        }
+        return this.renderFolderText(u, colors.lightBlue)
+      })
+    }
+
+    return (
+      <div style={{...commonStyles.clickable, display: 'flex', flexDirection: 'row'}} onClick={openFolder}>
+        {intersperseFn(i => this.renderFolderText(',', colors.lightTeal, i), folderText)}
+      </div>
+    )
+  }
+
+  render () {
+    const {folder: {isPublic, isEmpty, openFolder}} = this.props
+    const iconPath = folderIcon[isPublic ? 'public' : 'private'][isEmpty ? 'empty' : 'full']
+    return (
+      <div style={{display: 'flex', flexDirection: 'row', height: 40, alignItems: 'center'}}>
+        <div style={{...commonStyles.clickable, ...SVGFolderIcon(iconPath), marginRight: 6}} onClick={openFolder}/>
+        {this.renderFolderName()}
+      </div>
+    )
+  }
+}
+
+class CollapsableFolderList extends Component {
```

Comment:
This is fine for now but might be a good target for a refactor later. The collapsable nature of it really doens't have anything to do with folders per-se so it could just be generalized. If we have another component that does that we might want to merge them, then you could just take a row count and iterate on the row items, then take in a props.children to put underneath